src/SQL/Parser.hs

```diff
@@ -36,6 +37,20 @@ sqlP = lexeme $ space
                 <* eof
 
 -- select
+distinctP :: Parser Distinct
+distinctP = lexeme $ distinctOnP <|> simpleDistinctP
+  where
+    distinctOnP     = lexeme $
+                        string' "distinct on"
+                        <*  space
+                        $>  DistinctOn
+                        <*> between (lexeme (char '(')) (lexeme (char ')')) (NE.sepBy1 (lexeme expressionP) (lexeme (char ',')))
+    simpleDistinctP = lexeme $
+                        string' "distinct"
+                        $> Distinct
```

Comment:
`lexeme`ising `distinctOnP` and `simpleDistinctP` is probably redundant since we're `lexeme`ising it at the top level